File: backend/prisma/seed.ts
import { PrismaClient, MuscleGroup } from "@prisma/client";

const prisma = new PrismaClient();

const exercises: Array<{ name: string; muscleGroup: MuscleGroup; equipment?: string }> = [
  { name: "Barbell Bench Press", muscleGroup: "CHEST", equipment: "Barbell" },
  { name: "Incline Dumbbell Press", muscleGroup: "CHEST", equipment: "Dumbbell" },
  { name: "Decline Bench Press", muscleGroup: "CHEST", equipment: "Barbell" },
  { name: "Cable Fly", muscleGroup: "CHEST", equipment: "Cable" },
  { name: "Push-Up", muscleGroup: "CHEST" },
  { name: "Pec Deck", muscleGroup: "CHEST", equipment: "Machine" },
  { name: "Dumbbell Pullover", muscleGroup: "CHEST", equipment: "Dumbbell" },
  { name: "Landmine Press", muscleGroup: "CHEST", equipment: "Barbell" },
  { name: "Deadlift", muscleGroup: "BACK", equipment: "Barbell" },
  { name: "Barbell Row", muscleGroup: "BACK", equipment: "Barbell" },
  { name: "Chest Supported Row", muscleGroup: "BACK", equipment: "Machine" },
  { name: "Lat Pulldown", muscleGroup: "BACK", equipment: "Cable" },
  { name: "Pull-Up", muscleGroup: "BACK" },
  { name: "Seated Cable Row", muscleGroup: "BACK", equipment: "Cable" },
  { name: "Single Arm Dumbbell Row", muscleGroup: "BACK", equipment: "Dumbbell" },
  { name: "T-Bar Row", muscleGroup: "BACK", equipment: "Barbell" },
  { name: "Overhead Press", muscleGroup: "SHOULDERS", equipment: "Barbell" },
  { name: "Dumbbell Shoulder Press", muscleGroup: "SHOULDERS", equipment: "Dumbbell" },
  { name: "Lateral Raise", muscleGroup: "SHOULDERS", equipment: "Dumbbell" },
  { name: "Cable Lateral Raise", muscleGroup: "SHOULDERS", equipment: "Cable" },
  { name: "Rear Delt Fly", muscleGroup: "SHOULDERS", equipment: "Machine" },
  { name: "Face Pull", muscleGroup: "SHOULDERS", equipment: "Cable" },
  { name: "Arnold Press", muscleGroup: "SHOULDERS", equipment: "Dumbbell" },
  { name: "Upright Row", muscleGroup: "SHOULDERS", equipment: "Barbell" },
  { name: "Barbell Curl", muscleGroup: "BICEPS", equipment: "Barbell" },
  { name: "EZ Bar Curl", muscleGroup: "BICEPS", equipment: "EZ Bar" },
  { name: "Alternating Dumbbell Curl", muscleGroup: "BICEPS", equipment: "Dumbbell" },
  { name: "Hammer Curl", muscleGroup: "BICEPS", equipment: "Dumbbell" },
  { name: "Preacher Curl", muscleGroup: "BICEPS", equipment: "Machine" },
  { name: "Cable Curl", muscleGroup: "BICEPS", equipment: "Cable" },
  { name: "Concentration Curl", muscleGroup: "BICEPS", equipment: "Dumbbell" },
  { name: "Spider Curl", muscleGroup: "BICEPS", equipment: "Dumbbell" },
  { name: "Close Grip Bench Press", muscleGroup: "TRICEPS", equipment: "Barbell" },
  { name: "Tricep Pushdown", muscleGroup: "TRICEPS", equipment: "Cable" },
  { name: "Overhead Tricep Extension", muscleGroup: "TRICEPS", equipment: "Dumbbell" },
  { name: "Skull Crusher", muscleGroup: "TRICEPS", equipment: "EZ Bar" },
  { name: "Dips", muscleGroup: "TRICEPS" },
  { name: "Rope Pushdown", muscleGroup: "TRICEPS", equipment: "Cable" },
  { name: "Kickback", muscleGroup: "TRICEPS", equipment: "Dumbbell" },
  { name: "JM Press", muscleGroup: "TRICEPS", equipment: "Barbell" },
  { name: "Back Squat", muscleGroup: "LEGS", equipment: "Barbell" },
  { name: "Front Squat", muscleGroup: "LEGS", equipment: "Barbell" },
  { name: "Leg Press", muscleGroup: "LEGS", equipment: "Machine" },
  { name: "Romanian Deadlift", muscleGroup: "LEGS", equipment: "Barbell" },
  { name: "Hack Squat", muscleGroup: "LEGS", equipment: "Machine" },
  { name: "Leg Extension", muscleGroup: "LEGS", equipment: "Machine" },
  { name: "Leg Curl", muscleGroup: "LEGS", equipment: "Machine" },
  { name: "Walking Lunge", muscleGroup: "LEGS", equipment: "Dumbbell" },
  { name: "Bulgarian Split Squat", muscleGroup: "LEGS", equipment: "Dumbbell" },
  { name: "Goblet Squat", muscleGroup: "LEGS", equipment: "Dumbbell" },
  { name: "Hip Thrust", muscleGroup: "GLUTES", equipment: "Barbell" },
  { name: "Glute Bridge", muscleGroup: "GLUTES", equipment: "Barbell" },
  { name: "Cable Kickback", muscleGroup: "GLUTES", equipment: "Cable" },
  { name: "Step Up", muscleGroup: "GLUTES", equipment: "Dumbbell" },
  { name: "Sumo Deadlift", muscleGroup: "GLUTES", equipment: "Barbell" },
  { name: "Abductor Machine", muscleGroup: "GLUTES", equipment: "Machine" },
  { name: "Reverse Hyper", muscleGroup: "GLUTES", equipment: "Machine" },
  { name: "Single Leg Hip Thrust", muscleGroup: "GLUTES" },
  { name: "Crunch", muscleGroup: "CORE" },
  { name: "Cable Crunch", muscleGroup: "CORE", equipment: "Cable" },
  { name: "Hanging Leg Raise", muscleGroup: "CORE" },
  { name: "Plank", muscleGroup: "CORE" },
  { name: "Ab Wheel Rollout", muscleGroup: "CORE", equipment: "Ab Wheel" },
  { name: "Russian Twist", muscleGroup: "CORE", equipment: "Plate" },
  { name: "Pallof Press", muscleGroup: "CORE", equipment: "Cable" },
  { name: "Side Plank", muscleGroup: "CORE" },
  { name: "Treadmill Run", muscleGroup: "CARDIO", equipment: "Treadmill" },
  { name: "Elliptical", muscleGroup: "CARDIO", equipment: "Machine" },
  { name: "Bike Sprint", muscleGroup: "CARDIO", equipment: "Bike" },
  { name: "Rowing Machine", muscleGroup: "CARDIO", equipment: "Rower" },
  { name: "Jump Rope", muscleGroup: "CARDIO" },
  { name: "Stair Climber", muscleGroup: "CARDIO", equipment: "Machine" },
  { name: "HIIT Intervals", muscleGroup: "CARDIO" },
  { name: "Sled Push", muscleGroup: "CARDIO", equipment: "Sled" },
  { name: "Clean and Press", muscleGroup: "FULL_BODY", equipment: "Barbell" },
  { name: "Snatch", muscleGroup: "FULL_BODY", equipment: "Barbell" },
  { name: "Thruster", muscleGroup: "FULL_BODY", equipment: "Barbell" },
  { name: "Kettlebell Swing", muscleGroup: "FULL_BODY", equipment: "Kettlebell" },
  { name: "Burpee", muscleGroup: "FULL_BODY" },
  { name: "Farmer Carry", muscleGroup: "FULL_BODY", equipment: "Dumbbell" },
  { name: "Turkish Get Up", muscleGroup: "FULL_BODY", equipment: "Kettlebell" },
  { name: "Wall Ball", muscleGroup: "FULL_BODY", equipment: "Medicine Ball" },
];

async function main() {
  for (const exercise of exercises) {
    await prisma.exercise.upsert({
      where: { id: `${exercise.name}-${exercise.muscleGroup}`.toLowerCase().replace(/\s+/g, "-") },
      update: {},
      create: {
        id: `${exercise.name}-${exercise.muscleGroup}`.toLowerCase().replace(/\s+/g, "-"),
        ...exercise,
        isCustom: false,
      },
    });
  }
}

main()
  .then(() => prisma.$disconnect())
  .catch(async (error) => {
    console.error(error);
    await prisma.$disconnect();
    process.exit(1);
  });

File: backend/seed.py
"""
seed.py — Seed the exercise library

Run with: python seed.py

This populates the exercises table with 80+ exercises across all muscle groups.
These are global exercises (user_id = None) available to all users.
"""

import asyncio
import os
from sqlalchemy.ext.asyncio import create_async_engine, AsyncSession, async_sessionmaker
from sqlalchemy import select
from dotenv import load_dotenv

load_dotenv()

from app.models.exercise import Exercise, MuscleGroup
from app.database import Base

DATABASE_URL = os.getenv("DATABASE_URL", "postgresql+asyncpg://localhost/gymchad")

EXERCISES = [
    # ── CHEST ─────────────────────────────────────────────────────────────
    ("Barbell Bench Press", MuscleGroup.CHEST, "Barbell"),
    ("Incline Barbell Bench Press", MuscleGroup.CHEST, "Barbell"),
    ("Decline Barbell Bench Press", MuscleGroup.CHEST, "Barbell"),
    ("Dumbbell Bench Press", MuscleGroup.CHEST, "Dumbbell"),
    ("Incline Dumbbell Press", MuscleGroup.CHEST, "Dumbbell"),
    ("Dumbbell Fly", MuscleGroup.CHEST, "Dumbbell"),
    ("Cable Fly", MuscleGroup.CHEST, "Cable"),
    ("Pec Deck (Machine Fly)", MuscleGroup.CHEST, "Machine"),
    ("Dips (Chest)", MuscleGroup.CHEST, "Bodyweight"),
    ("Push-Up", MuscleGroup.CHEST, "Bodyweight"),
    ("Close-Grip Push-Up", MuscleGroup.CHEST, "Bodyweight"),

    # ── BACK ──────────────────────────────────────────────────────────────
    ("Deadlift", MuscleGroup.BACK, "Barbell"),
    ("Barbell Row", MuscleGroup.BACK, "Barbell"),
    ("T-Bar Row", MuscleGroup.BACK, "Barbell"),
    ("Pull-Up", MuscleGroup.BACK, "Bodyweight"),
    ("Chin-Up", MuscleGroup.BACK, "Bodyweight"),
    ("Lat Pulldown", MuscleGroup.BACK, "Cable"),
    ("Seated Cable Row", MuscleGroup.BACK, "Cable"),
    ("Single-Arm Dumbbell Row", MuscleGroup.BACK, "Dumbbell"),
    ("Cable Straight-Arm Pulldown", MuscleGroup.BACK, "Cable"),
    ("Rack Pull", MuscleGroup.BACK, "Barbell"),
    ("Meadows Row", MuscleGroup.BACK, "Barbell"),

    # ── SHOULDERS ─────────────────────────────────────────────────────────
    ("Overhead Press (Barbell)", MuscleGroup.SHOULDERS, "Barbell"),
    ("Seated Dumbbell Press", MuscleGroup.SHOULDERS, "Dumbbell"),
    ("Arnold Press", MuscleGroup.SHOULDERS, "Dumbbell"),
    ("Lateral Raise", MuscleGroup.SHOULDERS, "Dumbbell"),
    ("Cable Lateral Raise", MuscleGroup.SHOULDERS, "Cable"),
    ("Face Pull", MuscleGroup.SHOULDERS, "Cable"),
    ("Rear Delt Fly", MuscleGroup.SHOULDERS, "Dumbbell"),
    ("Upright Row", MuscleGroup.SHOULDERS, "Barbell"),
    ("Machine Shoulder Press", MuscleGroup.SHOULDERS, "Machine"),
    ("Push Press", MuscleGroup.SHOULDERS, "Barbell"),

    # ── BICEPS ────────────────────────────────────────────────────────────
    ("Barbell Curl", MuscleGroup.BICEPS, "Barbell"),
    ("Dumbbell Curl", MuscleGroup.BICEPS, "Dumbbell"),
    ("Hammer Curl", MuscleGroup.BICEPS, "Dumbbell"),
    ("Incline Dumbbell Curl", MuscleGroup.BICEPS, "Dumbbell"),
    ("Cable Curl", MuscleGroup.BICEPS, "Cable"),
    ("Preacher Curl", MuscleGroup.BICEPS, "Barbell"),
    ("Spider Curl", MuscleGroup.BICEPS, "Dumbbell"),
    ("Concentration Curl", MuscleGroup.BICEPS, "Dumbbell"),

    # ── TRICEPS ───────────────────────────────────────────────────────────
    ("Tricep Pushdown (Cable)", MuscleGroup.TRICEPS, "Cable"),
    ("Overhead Tricep Extension", MuscleGroup.TRICEPS, "Dumbbell"),
    ("Skull Crusher", MuscleGroup.TRICEPS, "Barbell"),
    ("Close-Grip Bench Press", MuscleGroup.TRICEPS, "Barbell"),
    ("Dips (Triceps)", MuscleGroup.TRICEPS, "Bodyweight"),
    ("Diamond Push-Up", MuscleGroup.TRICEPS, "Bodyweight"),
    ("Rope Pushdown", MuscleGroup.TRICEPS, "Cable"),
    ("Kickback", MuscleGroup.TRICEPS, "Dumbbell"),

    # ── LEGS ──────────────────────────────────────────────────────────────
    ("Barbell Squat", MuscleGroup.LEGS, "Barbell"),
    ("Front Squat", MuscleGroup.LEGS, "Barbell"),
    ("Hack Squat", MuscleGroup.LEGS, "Machine"),
    ("Leg Press", MuscleGroup.LEGS, "Machine"),
    ("Bulgarian Split Squat", MuscleGroup.LEGS, "Dumbbell"),
    ("Romanian Deadlift", MuscleGroup.LEGS, "Barbell"),
    ("Leg Curl (Lying)", MuscleGroup.LEGS, "Machine"),
    ("Leg Curl (Seated)", MuscleGroup.LEGS, "Machine"),
    ("Leg Extension", MuscleGroup.LEGS, "Machine"),
    ("Sumo Deadlift", MuscleGroup.LEGS, "Barbell"),
    ("Goblet Squat", MuscleGroup.LEGS, "Dumbbell"),
    ("Walking Lunges", MuscleGroup.LEGS, "Dumbbell"),
    ("Calf Raise (Standing)", MuscleGroup.LEGS, "Machine"),
    ("Calf Raise (Seated)", MuscleGroup.LEGS, "Machine"),

    # ── GLUTES ────────────────────────────────────────────────────────────
    ("Hip Thrust (Barbell)", MuscleGroup.GLUTES, "Barbell"),
    ("Hip Thrust (Machine)", MuscleGroup.GLUTES, "Machine"),
    ("Cable Kickback", MuscleGroup.GLUTES, "Cable"),
    ("Glute Bridge", MuscleGroup.GLUTES, "Bodyweight"),
    ("Donkey Kick", MuscleGroup.GLUTES, "Bodyweight"),
    ("Abductor Machine", MuscleGroup.GLUTES, "Machine"),

    # ── CORE ──────────────────────────────────────────────────────────────
    ("Plank", MuscleGroup.CORE, "Bodyweight"),
    ("Cable Crunch", MuscleGroup.CORE, "Cable"),
    ("Hanging Leg Raise", MuscleGroup.CORE, "Bodyweight"),
    ("Ab Rollout", MuscleGroup.CORE, "Equipment"),
    ("Russian Twist", MuscleGroup.CORE, "Bodyweight"),
    ("Crunch", MuscleGroup.CORE, "Bodyweight"),
    ("Decline Sit-Up", MuscleGroup.CORE, "Bodyweight"),
    ("Woodchopper", MuscleGroup.CORE, "Cable"),

    # ── CARDIO ────────────────────────────────────────────────────────────
    ("Treadmill Run", MuscleGroup.CARDIO, "Machine"),
    ("Stationary Bike", MuscleGroup.CARDIO, "Machine"),
    ("Rowing Machine", MuscleGroup.CARDIO, "Machine"),
    ("Stairmaster", MuscleGroup.CARDIO, "Machine"),
    ("Elliptical", MuscleGroup.CARDIO, "Machine"),
    ("Battle Ropes", MuscleGroup.CARDIO, "Equipment"),
    ("Jump Rope", MuscleGroup.CARDIO, "Equipment"),
    ("Box Jump", MuscleGroup.CARDIO, "Bodyweight"),

    # ── FULL BODY ─────────────────────────────────────────────────────────
    ("Barbell Clean", MuscleGroup.FULL_BODY, "Barbell"),
    ("Power Clean", MuscleGroup.FULL_BODY, "Barbell"),
    ("Kettlebell Swing", MuscleGroup.FULL_BODY, "Kettlebell"),
    ("Thruster", MuscleGroup.FULL_BODY, "Barbell"),
    ("Burpee", MuscleGroup.FULL_BODY, "Bodyweight"),
    ("Turkish Get-Up", MuscleGroup.FULL_BODY, "Kettlebell"),
]


async def seed():
    engine = create_async_engine(DATABASE_URL, echo=False)
    async with engine.begin() as conn:
        await conn.run_sync(Base.metadata.create_all)

    session_factory = async_sessionmaker(engine, class_=AsyncSession, expire_on_commit=False)

    async with session_factory() as session:
        # Check if already seeded
        result = await session.execute(select(Exercise).limit(1))
        if result.scalar_one_or_none():
            print("Database already has exercises — skipping seed.")
            return

        exercises = [
            Exercise(name=name, muscle_group=group, equipment=equipment, is_custom=False)
            for name, group, equipment in EXERCISES
        ]
        session.add_all(exercises)
        await session.commit()
        print(f"✅ Seeded {len(exercises)} exercises")

    await engine.dispose()


if __name__ == "__main__":
    asyncio.run(seed())
